### ISSUE TITLE
ERROR: Can't install RMagick 4.2.6. Can't find ImageMagick with pkg-config

### ISSUE THREAD

**Issue by apurvis** [rmagick/rmagick#1357]
### Description
Failed to install `RMagick` on macOS Monterey with up to date `imagemagick` and `pkg-info` claiming it can't find `ImageMagick`.

Logs at `.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/extensions/arm64-darwin-21/3.1.0/rmagick-4.2.4/mkmf.log` contents:
```
find_executable: checking for brew... -------------------- yes

--------------------

assert_minimum_ruby_version!: checking for Ruby version >= 2.3.0... -------------------- yes

--------------------

find_executable: checking for pkg-config... -------------------- yes

--------------------

ERROR: Can't install RMagick 4.2.4. Can't find ImageMagick with pkg-config
```

### Solution
I managed to figure this out before hitting "submit new issue" so I'm just posting this because it was an annoying thing to figure out and someone else is prolly going to run into it.

ruby was whining about `pkg-info` not providing a path, which was weird because it seemed to be working fine:

```bash
$ pkg-config --cflags imagemagick
-Xpreprocessor -fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/opt/homebrew/Cellar/imagemagick/7.1.0-49_1/include/ImageMagick-7
$ pkg-config --libs imagemagick
-L/opt/homebrew/Cellar/imagemagick/7.1.0-49_1/lib -lMagickCore-7.Q16HDRI
```
except then I saw
```bash
$ pkg-config --list-all
Segmentation fault: 11
```
turned out it was a weird issue in a config file for `jpeg-xl` installed by homebrew, specifically this one: `/opt/homebrew/Cellar/jpeg-xl/0.7.0/lib/pkgconfig/libhwy-test.pc`.  

contents:
```
prefix=/opt/homebrew/Cellar/jpeg-xl/0.7.0
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: libhwy-test
Description: Efficient and performance-portable SIMD wrapper, test helpers.
Requires: gtest
Version: 1.0.1
Libs: -L${libdir} -lhwy_test
Cflags: -I${includedir}
```

`libdir=${exec_prefix}/lib` needs to become `libdir=${prefix}/lib`

### System Configuration
macOS

- ImageMagick version: 7.1.0
- RMagick version: saw it with 4.2.4 and 4.3.0
- Ruby version: 3.1.2
- Environment (Operating system, version and so on): macOS
- Additional information:

**Comment by aliken9** [comment]
My file was in a different spot, but this command helped me figure it out: `pkg-config --debug --list-all`

**Comment by dlemstra** [comment]
Looks like this was fixed in the following commit: https://github.com/google/highway/commit/357e21beabb1af037f20130b4195fa5d0e6bbbfb. Closing this issue because this is in a dependency issue and not an rmagick/imagemagick  issue.
